mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -1,409 +1,97 @@
-            :param pl_module: pytorch lightning base module
-            """
-            # If validation loop is enabled (meaning `validation_step` is overridden),
-            # log metrics in `on_validaion_epoch_end` to avoid logging the same metrics
-            # records twice
-            if not trainer.enable_validation:
-                self._log_epoch_metrics(trainer, pl_module)
-
-        @rank_zero_only
-        def on_validation_epoch_end(self, trainer, pl_module):
-            """
-            Log loss and other metrics values after each validation epoch
-
-            :param trainer: pytorch lightning trainer instance
-            :param pl_module: pytorch lightning base module
-            """
-            self._log_epoch_metrics(trainer, pl_module)
-
-    else:
-
-        @rank_zero_only
-        def on_epoch_end(self, trainer, pl_module):
-            """
-            Log loss and other metrics values after each epoch
-
-            :param trainer: pytorch lightning trainer instance
-            :param pl_module: pytorch lightning base module
-            """
-            self._log_epoch_metrics(trainer, pl_module)
-
-    @rank_zero_only
-    def on_train_batch_end(
-        self, trainer, pl_module, *args
-    ):  # pylint: disable=signature-differs,arguments-differ,unused-argument
-        """
-        Log metric values after each step
-
-        :param trainer: pytorch lightning trainer instance
-        :param pl_module: pytorch lightning base module
-        """
-        if not self.log_every_n_step:
-            return
-        # When logging at the end of a batch step, we only want to log metrics that are logged
-        # on steps. For forked metrics (metrics logged on both steps and epochs), we exclude the
-        # metric with the non-forked name (eg. "loss" when we have "loss", "loss_step" and
-        # "loss_epoch") so that this is only logged on epochs. We also record which metrics
-        # we've logged per step, so we can later exclude these from metrics logged on epochs.
-        metrics = _get_step_metrics(trainer)
-        metric_items = [
-            (name, val)
-            for (name, val) in metrics.items()
-            if (name not in self._epoch_metrics) and (f"{name}_step" not in metrics.keys())
+    met = Metric(key=key, value=value, timestamp=time, step=step)
+    with _metric_queue_lock:
+        _metric_queue.append((run_id, met))
+        if len(_metric_queue) > _MAX_METRIC_QUEUE_SIZE:
+            _thread_pool.submit(_flush_queue)
+
+
+FAILED = object()
+autolog_run = None
+
+
+def _get_run():
+    global autolog_run
+    if autolog_run is None:
+        autolog_run = mlflow.active_run()
+    if autolog_run is None:
+        try:
+            autolog_run = mlflow.start_run()
+        except Exception:
+            # don't try again.
+            autolog_run = FAILED
+    if autolog_run not in (None, FAILED):
+        return autolog_run
+    return None
```

Comment:
Why do we need this function? Can we just use `mlflow.active_run`?

Reply:
Pretty sure this is copied from 1.x branch, but can double check

Reply:
so -- who would be starting the autolog run if we drop this function?

Reply:
> Pretty sure this is copied from 1.x branch, but can double check

Can you share the permalink?

Reply:
sorry, i must have been mistaken, not sure where this is from. But the issue remains -- if we remove it, who starts the run?

Reply:
> so -- who would be starting the mlflow run if we drop this function?

`safe_patch` handles that.

```
def safe_patch(
    autologging_integration, destination, function_name, patch_function, manage_run=False
):
    """
    Patches the specified `function_name` on the specified `destination` class for autologging
    purposes, preceding its implementation with an error-safe copy of the specified patch
    `patch_function` with the following error handling behavior:
        - Exceptions thrown from the underlying / original function
          (`<destination>.<function_name>`) are propagated to the caller.
        - Exceptions thrown from other parts of the patched implementation (`patch_function`)
          are caught and logged as warnings.
    :param autologging_integration: The name of the autologging integration associated with the
                                    patch.
    :param destination: The Python class on which the patch is being defined.
    :param function_name: The name of the function to patch on the specified `destination` class.
    :param patch_function: The patched function code to apply. This is either a `PatchFunction`
                           class definition or a function object. If it is a function object, the
                           first argument should be reserved for an `original` method argument
                           representing the underlying / original function. Subsequent arguments
                           should be identical to those of the original function being patched.
    :param manage_run: If `True`, applies the `with_managed_run` wrapper to the specified
                       `patch_function`, which automatically creates & terminates an MLflow
                       active run during patch code execution if necessary. If `False`,
                       does not apply the `with_managed_run` wrapper to the specified
                       `patch_function`.
    """
```

Reply:
i see, sorry I missed that. I'll remove the function in favor of active_run.

Reply:
updated.